### PR TITLE
Problem: zmq_poller_[add/modify] accept invalid events arguments silently

### LINKED ISSUE #3088: zmq_poller_add/zmq_poller_modify should reject invalid events arguments
*Please use this template for reporting suspected bugs or requests for help.*

# Issue description

zmq_poller_add/zmq_poller_modify should reject invalid events arguments, rather than accepting them silently.

# Environment

* libzmq version (commit hash if unreleased): 4.2.5
* OS: any

# Minimal test code / Steps to reproduce the issue

```
void *ctx = zmq_ctx_new();
void *socket = zmq_socket(ctx, ZMQ_DEALER);
void *poller = zmq_poller_new();
int res = zmq_poller_add(poller, socket, nullptr, 42);
```

# What's the actual result? (include assertion message & call stack if applicable)

`res == 0`

# What's the expected result?

`res == -1 && zmq_errno() == EINVAL`

### PR DESCRIPTION
**Pull request by sigiesec**
Solution: check and return an error on invalid arguments. Fixes #3088

### CODE REVIEW

File: include/zmq.h [hunk 1]
```diff
@@ -463,14 +463,16 @@ ZMQ_EXPORT int zmq_socket_monitor (void *s, const char *addr, int events);
 #define ZMQ_POLLERR 4
 #define ZMQ_POLLPRI 8
 
-typedef struct zmq_pollitem_t
-{
-    void *socket;
 #if defined _WIN32
-    SOCKET fd;
+typedef SOCKET zmq_fd_t;
```
Comment (bluca) [inline include/zmq.h]: I'm not sure I like the idea of adding a new typedef to the ABI, especially changing existing ones... it's fine for internal usage ofc

Comment (sigiesec) [reply include/zmq.h]: Why should a typedef change the ABI?

Comment (bluca) [reply include/zmq.h]: I meant changing existing definitions - with all the broken compilers out there (*cough*oracle*cough*) I'd really rather avoid the risk of changing existing structs when there's no overwhelming benefits

**Comment by sigiesec** [comment]
The test failure in the `BUILD_TYPE=default CURVE=libsodium DRAFT=enabled GSSAPI=enabled PGM=enabled NORM=enabled TIPC=enabled` build is strange. The tests that fail are deterministic. I will investigate that.

File: include/zmq.h (hunk 1 above)
Comment (sigiesec) [reply include/zmq.h]: Hm. Ok. I will revert the changes to zmq.h

**Comment by sigiesec** [comment]
Ah, I think I got it. Strange that the other test runs are successful, errno was not set at all.

**Pull request merged**